PrincessPaperplane/discordBot.py

```diff
@@ -304,89 +316,97 @@ async def on_message(message):
-                ext = "&test"
-            url = "https://501-legion.de/princesspaperplane/generateLevel.php?user=%s&time=%s%s" % (author.id, time.time(), ext)
+    if server.id == LIVE_SERVER:
+        cur.execute("SELECT exp, level, name FROM user_info WHERE id = %s", (author.id,))
+    if server.id == TEST_SERVER:
+        cur.execute("SELECT exp, level, name FROM user_info_test WHERE id = %s", (author.id,))
+    row = cur.fetchone()
 
-            # embed current XP, level and missing XP for next levelup
-            title = "Dein aktuelles Level: %d" % (level)
-            description = "Aktuelle XP: %d\nVerbleibende XP bis Level %d: %d" % (exp, nextLevel, expLeft)
-            colour = author.top_role.colour
+    if cur.rowcount == 0:
+        exp = 0
+        if server.id == LIVE_SERVER:
+            cur.execute("INSERT INTO user_info (`id`, `name`, `exp`, `expTime`, `avatar_url`) VALUES (%s, %s, %s, %s, %s)", (author.id, author.name, exp, time.time(), author.avatar_url, ))
+            cur.execute("SELECT exp, level, name FROM user_info WHERE id = %s", (author.id,))
+        if server.id == TEST_SERVER:
+            cur.execute("INSERT INTO user_info_test (`id`, `name`, `exp`, `expTime`, `avatar_url`) VALUES (%s, %s, %s, %s, %s)", (author.id, author.name, exp, time.time(), author.avatar_url, ))
+            cur.execute("SELECT exp, level, name FROM user_info_test WHERE id = %s", (author.id,))
+        row = cur.fetchone()
+
+    # store current variables
+    exp = row[0]
+    level = row[1]
+    nextLevel = level + 1
+    nextLevelUp = await getLevelUp(level)
+    expLeft = nextLevelUp - exp
+
+    # generate image with stats
+    ext = ""
+    if server.id == TEST_SERVER:
+        ext = "&test"
+    url = "https://501-legion.de/princesspaperplane/generateLevel.php?user=%s&time=%s%s" % (author.id, time.time(), ext)
 
-            embed = discord.Embed(title=title, description=description, colour=colour)
-            embed.set_author(name=author.name, icon_url=author.avatar_url_as(format="png"))
-            embed.set_image(url=url)
+    # embed current XP, level and missing XP for next levelup
+    title = "Dein aktuelles Level: %d" % (level)
+    description = "Aktuelle XP: %d\nVerbleibende XP bis Level %d: %d" % (exp, nextLevel, expLeft)
+    colour = author.top_role.colour
 
-            await channel.send(embed=embed)
+    embed = discord.Embed(title=title, description=description, colour=colour)
+    embed.set_author(name=author.name, icon_url=author.avatar_url_as(format="png"))
+    embed.set_image(url=url)
 
-        # handle command !w to roll a dice
-        if "w" == message.content[1]:
-            # check if !w-command is correct
-            try:
-                # dice type like W6
-                dice = message.content.split('!w')
+    await channel.send(embed=embed)
+            
 
-                # multiple dices
-                multiply = dice[1].split('x')
+@commands.command(aliases=['w','d'])

```

Comment:
Wirft den Fehler `discord.ext.commands.errors.CommandNotFound: Command "w" is not found`
Stattdessen hat `@bot.command` funktioniert